### PR TITLE
remove pulumi deplendency, use third_party instead

### PR DESCRIPTION
**Pull request by howieyuen**
kusion only use some util funcs of `pulumi`, we can make a copy of utils, save it in third party.

close: https://github.com/KusionStack/kusion/issues/58

### CODE REVIEW

File: pkg/scaffold/external_templates.go [hunk 1]
```diff
@@ -318,7 +320,7 @@ func newTemplateNotFoundError(templateDir string, templateName string) error {
 	// Attempt to read the directory to offer suggestions.
 	infos, err := ioutil.ReadDir(templateDir)
 	if err != nil {
-		contract.IgnoreError(err)
+		log.Infof("ioutil.ReadDir(%s) error: %v", templateDir, err)
```
Comment (elliotxx) [inline pkg/scaffold/external_templates.go:323]: Consider using `Warnf` output for ignored errors

Comment (howieyuen) [reply pkg/scaffold/external_templates.go:323]: ok, thanks for tips

**Review by chai2010 — APPROVED** [review]
LGTM

**Review by elliotxx — APPROVED** [review]
lgtm

**Pull request merged**